numpy/core/src/umath/loops_arithmetic.dispatch.c.src

```diff
@@ -20,11 +20,69 @@
 //###############################################################################
 /********************************************************************************
  ** Defining the SIMD kernels
+ *
+ * Floor division of signed is based on T. Granlund and P. L. Montgomery
+ * “Division by invariant integers using multiplication(see [Figure 6.1]
+ * http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.1.2556)"
+ * For details on TRUNC division see simd/intdiv.h for more clarification
+ ***********************************************************************************
+ ** Figure 6.1: Signed division by run–time invariant divisor, rounded towards -INF
+ ***********************************************************************************
+ * For q = FLOOR(a/d), all sword:
+ *     sword −dsign = SRL(d, N − 1);
+ *     uword −nsign = (n < −dsign);
+ *     uword −qsign = EOR(−nsign, −dsign);
+ *     q = TRUNC((n − (−dsign ) + (−nsign))/d) − (−qsign);
  ********************************************************************************/
 #if NPY_SIMD
 /**begin repeat
- *  #sfx = u8, u16, u32, u64#
+ * #sfx    = u8,  u16, u32, u64, s8, s16, s32, s64#
+ * #len    = 8,   16,  32,  64,  8,  16,  32,  64#
+ * #signed = 0*4, 1*4#
  */
+#if @signed@
+static NPY_INLINE npyv_@sfx@
+simd_floor_divide_@sfx@(npyv_lanetype_@sfx@ *src, const npyv_@sfx@x3 divisor, npyv_lanetype_@sfx@ scalar)
+{
+    npyv_@sfx@ a, nsign_d, nsign_a, diff_sign, to_ninf, trunc, floor;
+    npyv_b@len@ greater_min, noverflow;
+
+    nsign_d   = npyv_setall_@sfx@(scalar < 0);
+    a         = npyv_load_@sfx@(src);
+    nsign_a   = npyv_cvt_@sfx@_b@len@(npyv_cmplt_@sfx@(a, nsign_d));
+    nsign_a   = npyv_and_@sfx@((npyv_@sfx@)nsign_a, npyv_setall_@sfx@(1));
+    diff_sign = npyv_sub_@sfx@((npyv_@sfx@)nsign_a, nsign_d);
+    to_ninf   = npyv_xor_@sfx@((npyv_@sfx@)nsign_a, nsign_d);
+    trunc     = npyv_divc_@sfx@(npyv_add_@sfx@(a, diff_sign), divisor);
+
+    if (NPY_UNLIKELY(-1 == scalar)) {
+        greater_min = npyv_cmpgt_@sfx@(a, npyv_setall_@sfx@(NPY_MIN_INT@len@));
+        noverflow   = npyv_cvt_b@len@_@sfx@(npyv_setall_@sfx@(-1));
+        noverflow   = npyv_and_b@len@(noverflow, greater_min);
+        if (npyv_tobits_b@len@(noverflow) != (((npy_uint64)1) << npyv_nlanes_@sfx@)-1) {
```

Comment:
This overflows for 8 bits as nlanes value is 64, any workaround to directly use the value?

Reply:
Seems like CI failed for the same reason

Reply:
Thanks @seiko2plus , I have added the same, I used a `#if` here to avoid compiler warning, as it's not picking the path and assumes an overflow.